src/component/component.js

```diff
@@ -414,6 +413,13 @@ export function component<P, X, C>(opts : ComponentOptionsType<P, X, C>) : Compo
         });
 
         const clone = ({ decorate = identity } = {}) => {
+            // Delete any existence of the decorate function before clone the instance
+            // The _originalDefinition was added in the parent/props.js in the decorate logic
+            // $FlowFixMe
+            values(propsDef).forEach(prop => {
+                // $FlowFixMe
+                delete prop._originalDefinition;
```

Comment:
Not a big fan of making `clone` a special case where we do cleanup. There are plenty of other cases where the same component can be rendered multiple times. Also clone does not imply that the original component is not needed any more.